app/liquid_tags/unified_embed/tag.rb

```diff
@@ -54,12 +54,26 @@ def self.validate_link!(link)
       path = uri.path.presence || "/"
       response = http.request_head(path)
 
-      unless response.is_a?(Net::HTTPSuccess) || response.is_a?(Net::HTTPMovedPermanently)
+      case response
```

Comment:
After much discussion with @dscottS3, I decided NOT to recursively check redirects.

While testing, I realized that the validation of some redirect locations were returning statuses like `#<Net::HTTPMethodNotAllowed 405 Method Not Allowed readbody=true>`. This response would have failed my checks, even though the original URL embedded just fine.

So while it is possible that a redirected URL may not embed correctly, I believe the chances of that are small enough to ignore at this present time. Bear in mind that we are yet to implement the OpenGraph fallback, which will provide much better handling than what we have now.